### PR TITLE
proper checks for version and header

### LINKED ISSUE #2214: MEI header version interaction
If you remove the `meiHead` from the tree Verovio parses it as a MEI3 file regardless of what the `meiversion` attribute states.

```xml
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
   <music>
      <body>
         <mdiv>
            <score>
               <scoreDef>
                  <staffGrp>
                     <staffDef n="1" lines="5" clef.shape="C" clef.line="3" />
                  </staffGrp>
               </scoreDef>
               <section>
                  <measure right="dbl" n="1">
                     <staff n="1">
                        <layer n="1">
                           <note xml:id="note-01" dur="4" oct="4" pname="c" stem.dir="down" />
                           <note xml:id="note-02" dur="4" oct="4" pname="c" stem.dir="up" />
                           <note xml:id="note-03" dur="4" oct="4" pname="c" stem.dir="down" />
                           <note xml:id="note-04" dur="4" oct="4" pname="c" stem.dir="up" />
                        </layer>
                     </staff>
                     <mordent staff="1" startid="#note-01" />
                     <mordent staff="1" startid="#note-02" form="upper" />
                     <mordent staff="1" startid="#note-03" form="upper" long="true" />
                     <mordent staff="1" startid="#note-04" place="below" form="upper" long="true" />
                  </measure>
               </section>
            </score>
         </mdiv>
      </body>
   </music>
</mei>
```

### PR DESCRIPTION
**Pull request by rettinghaus**
fixes #2214

### CODE REVIEW

File: src/iomei.cpp [hunk 1]
```diff
@@ -2893,26 +2893,35 @@ bool MEIInput::ReadDoc(pugi::xml_node root)
...
+        if (version == "5.0.0-dev")
+            m_version = MEI_5_0_0_dev;
+        else if (version == "4.0.1")
+            m_version = MEI_4_0_1;
+        else if (version == "4.0.0")
+            m_version = MEI_4_0_0;
+        else if (version == "3.0.0")
+            m_version = MEI_3_0_0;
+        else if (version == "2013")
+            m_version = MEI_2013;
```
Comment (lpugin) [inline src/iomei.cpp:2907]: Please add `{}`

File: src/iomei.cpp [hunk 2]
```diff
@@ -2893,26 +2893,35 @@ bool MEIInput::ReadDoc(pugi::xml_node root)
...
-                m_version = MEI_4_0_1;
-            else if (version == "4.0.0")
-                m_version = MEI_4_0_0;
-            else if (version == "3.0.0")
-                m_version = MEI_3_0_0;
-            else if (version == "2013")
-                m_version = MEI_2013;
+        // default to MEI 4
+        LogWarning("No meiversion found, falling back to MEI4");
+        m_version = MEI_4_0_0;
```
Comment (lpugin) [inline src/iomei.cpp:2912]: We need to fallback to `5.0.0-dev`

File: src/iomei.cpp (hunk 1 above)
Comment (rettinghaus) [reply src/iomei.cpp]: done

File: src/iomei.cpp [hunk 3]
```diff
@@ -2893,26 +2893,40 @@ bool MEIInput::ReadDoc(pugi::xml_node root)
...
+        }
+        else if (version == "4.0.0") {
+            m_version = MEI_4_0_0;
+        }
+        else if (version == "3.0.0") {
+            m_version = MEI_3_0_0;
+        }
+        else if (version == "2013") {
+            m_version = MEI_2013;
+        }
```
Comment (ahankinson) [inline src/iomei.cpp:2912]: Should this handle the case where it doesn't match any of these values?

Comment (rettinghaus) [reply src/iomei.cpp:2912]: I guess not. Haven't thought about it though, as I just separated the version check from the header handling.

File: src/iomei.cpp [hunk 4]
```diff
@@ -2893,26 +2893,40 @@ bool MEIInput::ReadDoc(pugi::xml_node root)
...
-                m_version = MEI_5_0_0_dev;
-            else if (version == "4.0.1")
-                m_version = MEI_4_0_1;
-            else if (version == "4.0.0")
-                m_version = MEI_4_0_0;
-            else if (version == "3.0.0")
-                m_version = MEI_3_0_0;
-            else if (version == "2013")
-                m_version = MEI_2013;
+        // default to MEI 4
```
Comment (musicEnfanthen) [inline src/iomei.cpp:2915]:
```suggestion
        // default to MEI 5
```

File: src/iomei.cpp (hunk 3 above)
Comment (rettinghaus) [reply src/iomei.cpp:2912]: By default setting `m_version` is `MEI_UNDEFINED`. I think it makes sense to go with that if we find a nonsense value in `meiversion`.

**Comment by rettinghaus** [comment]
@lpugin anything that needs to be change?

**Pull request merged**